account_currency_tools/wizard/exchange_realization.py

```diff
@@ -0,0 +1,960 @@
+            'account.account',
+            string="Loss Exchange Rate Account",
+            domain="[('type', '=', 'other')]"),
+        'skip_close_fiscalyear': fields.boolean(
+            'Skip Close Fiscal Year Check'),
+        'skip_opening_entry': fields.boolean(
+            'Skip Opening Journal Entry Check'),
+    }
+
+    _defaults = {
+        'company_id': _get_default_company,
+        'fiscalyear_id': _get_fiscalyear,
+        'state': 'draft',
+        'target_move': 'posted',
+    }
+
+    def get_account_balance(self, cr, uid, args, context=None):
+        query = '''
+            SELECT
+                aml.currency_id,
+                SUM(aml.debit - aml.credit) AS balance
+            FROM account_move_line AS aml
+            INNER JOIN account_period AS ap ON ap.id = aml.period_id
+            INNER JOIN account_move AS am ON am.id = aml.move_id
+            WHERE
+                aml.account_id IN (%(account_ids)s) AND
+                aml.currency_id IN (%(currency_ids)s) AND
+                aml.currency_id IS NOT NULL AND
+                aml.state <> 'draft' AND
+                am.state IN (%(states)s) AND
+                ap.id IN (%(period_ids)s)
+            GROUP BY aml.currency_id
+        ''' % args
+        cr.execute(query)
+        res = cr.dictfetchall()
+        return res
+
+    def get_values_from_aml(self, cr, uid, args, context=None):
+        query = '''
+            SELECT
+                aml.account_id,
+                aml.currency_id,
+                SUM(aml.amount_currency) AS foreign_balance,
+                SUM(aml.debit - aml.credit) AS balance
+            FROM account_move_line AS aml
+            INNER JOIN account_period AS ap ON ap.id = aml.period_id
+            INNER JOIN account_move AS am ON am.id = aml.move_id
+            WHERE
+                aml.account_id IN (%(account_ids)s) AND
+                aml.currency_id IS NOT NULL AND
+                aml.state <> 'draft' AND
+                am.state IN (%(states)s) AND
+                ap.id IN (%(period_ids)s)
+            GROUP BY aml.account_id, aml.currency_id
+        ''' % args
+        cr.execute(query)
+        res = cr.dictfetchall()
+        return res
+
+    def get_accounts_from_aml(self, cr, uid, args, context=None):
+        query = '''
+            SELECT
+                aml.account_id
+            FROM account_move_line AS aml
+            INNER JOIN account_account AS aa ON aa.id = aml.account_id
+            INNER JOIN account_period AS ap ON ap.id = aml.period_id
+            INNER JOIN account_move AS am ON am.id = aml.move_id
+            WHERE
+                aa.type = '%(account_type)s' AND
+                aml.currency_id IS NOT NULL AND
+                aml.state <> 'draft' AND
+                am.state IN (%(states)s) AND
+                aml.company_id = %(company_id)d AND
+                aa.parent_left BETWEEN %(parent_left)d AND %(parent_right)d AND
```

Comment:
@JulioSerna then after fixing sql injection offending code this PR can be merged

Regards.